spring-cloud-kubernetes-integration-tests/spring-cloud-kubernetes-client-reactive-discovery-client-it/src/main/java/org/springframework/cloud/kubernetes/client/reactive/discovery/it/KubernetesClientReactiveDiscoveryClientApplicationIt.java

```diff
@@ -32,21 +33,20 @@
 @RestController
 public class KubernetesClientReactiveDiscoveryClientApplicationIt {
 
-	@Autowired
-	KubernetesInformerReactiveDiscoveryClient discoveryClient;
+	private final KubernetesInformerReactiveDiscoveryClient discoveryClient;
+
+	public KubernetesClientReactiveDiscoveryClientApplicationIt(
+			KubernetesInformerReactiveDiscoveryClient discoveryClient) {
+		this.discoveryClient = discoveryClient;
+	}
 
 	public static void main(String[] args) {
 		SpringApplication.run(KubernetesClientReactiveDiscoveryClientApplicationIt.class, args);
 	}
 
 	@GetMapping("/services")
 	public Mono<String> services() {
-		return discoveryClient.getServices().reduce("", (a, b) -> {
```

Comment:
this is tricky :) it works in the _current_ version of Java (up to 17 I tested), but it still violates the rules of reduce, specifically this is not symmetric. But, the changes here are small, so not a big deal.